coff.ml

```diff
@@ -776,17 +846,31 @@ module Coff = struct
     ()
 
   let put oc x =
-    emit_int16 oc x.machine;
-
     let () =
       let no = ref 0 in
       List.iter
         (fun s -> incr no; assert(s.sec_pos < 0); s.sec_pos <- !no)
         x.sections
     in
 
-    emit_int16 oc (List.length x.sections);
-    emit_int32 oc x.date;
+    if x.bigobj then begin
+      emit_int16 oc 0; (* Sig1 *)
+      emit_int16 oc 0xffff; (* Sig2 *)
+      emit_int16 oc 2; (* Version *)
```

Comment:
Why is this written on this side (for bigobj) but not on the other (for the legacy one)?

There may well be a reason, I was just scratching my head trying to work it out!

Reply:
I address your other comments later (they're mostly cosmetic and I have nothing against them. I tried to change the original sources minimally while you're trying to make them more readable). Regarding this:
 https://github.com/alainfrisch/flexdll/blob/f40ed2d57bbb70ac8223d30a6ffa737f2c0e430e/coff.ml#L856-L873
I don't quite understand your question. Why Sig1, Sig2, and Version are not emitted for the standard COFF? Just because its header does not include these fields.

Reply:
This makes sense to me now w.r.t. the comment below - I hadn't registered properly that what we had before was an incorrect reading of an ECOFF header (assuming an import library)